src/library/autodj/autodjprocessor.cpp

```diff
@@ -1245,33 +1245,21 @@ void AutoDJProcessor::calculateTransition(DeckAttributes* pFromDeck,
         // If only the outro or intro length is marked but not both, use the one
         // that is marked for the transition time. If neither is marked, fall
         // back to the transition time from the spinbox.
-        if (outroLength > 0 && introLength > 0) {
-            if (outroLength > introLength) {
-                pFromDeck->fadeBeginPos = outroEnd - introLength;
-                pFromDeck->fadeEndPos = outroEnd;
-                pToDeck->startPos = introStart;
-            } else {
-                pFromDeck->fadeBeginPos = outroStart;
-                pFromDeck->fadeEndPos = outroEnd;
-                pToDeck->startPos = introStart;
+        if (outroLength > 0) {
+            if (introLength <= 0 || introLength > outroLength) {
+                // no intro defined, use outro lengh.
```

Comment:
typo

Reply:
`introLength > outroLength` implies "no intro defined"? I don't understand this reasoning.

Reply:
```suggestion
                // Use outro length when the intro is not defined or longer than the outro. 
```

Reply:
@daschuer ping